src/libutil/filesystem.cpp

```diff
@@ -39,6 +39,7 @@
 #include <boost/filesystem.hpp>
 #include <boost/foreach.hpp>
 #include <boost/regex.hpp>
+#include <boost/shared_ptr.hpp>
```

Comment:
Instead of boost/shared_ptr.hpp, let's include OpenImageIO/refcnt.h, which defines an OIIO::shared_ptr<> that underneath is std::shared_ptr on C++11 and uses boost::shared_ptr otherwise (and for each case, includes the right headers).